libchisel/src/deployer.rs

```diff
@@ -5,6 +5,33 @@ use super::*;
 use byteorder::{LittleEndian, WriteBytesExt};
 use rustc_hex::FromHex;
 
+pub enum DeployerType {
+    Memory,
+    CustomSection,
+}
+
+/// Struct on which ModuleValidator is implemented.
+pub struct Deployer {
+    pub deployer_type: DeployerType,
+    pub payload: Vec<u8>,
```

Comment:
I think it looks cleaner to use just the structured enum pattern, such that DeployerType variants will hold the payload.

Reply:
> I think it looks cleaner to use just the structured enum pattern, such that DeployerType variants will hold the payload.

Can you give an example how you imagine the API?

> Also, could you use a slice member with lifetime parameters instead of Vec?

Can you also give an example?